odoo/models.py

```diff
@@ -4859,7 +4859,8 @@ def _mapped_func(self, func):
         if self:
             vals = [func(rec) for rec in self]
             if isinstance(vals[0], BaseModel):
-                return vals[0].union(*vals)         # union of all recordsets
+                prefetch = vals[0]._prefetch
+                return vals[0].union(*vals).with_prefetch(prefetch)  # union of all recordsets
```

Comment:
Could this go in `union` instead? Union could automatically preserve the `prefetch` if all records have the same `_prefetch`.

Reply:
Initially, the fix was in the union method. I finally moved it here in order to limit the impact of this change. 
However, it would seem logical that the union method also takes care of the prefetch informations.

Reply:
`union` does not aim at transferring the prefetch object from input values to output value.
And "if all records have the same `_prefetch`" is too much of a special case.